mlx-nn/src/utils.rs

```diff
@@ -0,0 +1,118 @@
+//! Utility types and functions.
+
+use std::rc::Rc;
+
+use mlx_rs::module::FlattenedModuleParam;
+use mlx_rs::Array;
+
+/// A custom type to indicate whether a `Module` should include a bias or not.
+/// Default to `Yes`.
+#[derive(Debug, Clone, Copy, Default)]
+pub enum WithBias {
```

Comment:
Will be nice to eventually be able to define bias as `Tensor<Shape>`.. I wonder if there's some way we might already be able to do something similar with Array

Reply:
This is a part that I'm not too sure about. This, on the on hand, is not consistent with how we handle optional arg in other `Module`s or functions, but on the other hand the bias would either be `zeros` or `random::uniform`. But we could probably get rid of this and just define an associated `pub const` for the conv layers and linear layers (these are the only `Module`s that use this type right now